java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/protocol/NbCodeClientWrapper.java

```diff
@@ -144,4 +144,19 @@ public void semanticHighlighting(SemanticHighlightingParams params) {
     public void notifyProgress(ProgressParams params) {
         remote.notifyProgress(params);
     }
+
+    @Override
+    public CompletableFuture<String> createTextEditorDecoration(DecorationRenderOptions params) {
```

Comment:
I am not sure I understand the purpose of these new messages. They look standard (e.g. `window/createTextEditorDecoration`)...

Reply:
Yes, they are extensions for VSCode.

Reply:
Decorations are not part of the standard LSP. We need to extend it.

Reply:
Then it is surprising we put the messages into the `window/` namespace. But I see that we have done that even before with `window/showInputBox` & co.